coremltools/converters/mil/frontend/torch/torchir_passes.py

```diff
@@ -118,12 +118,14 @@ def _construct_nodes_to_fuse_inputs(nodes_to_fuse):
                 kind = "_internal_op_tensor_inplace_fill"
 
             nodes_to_fuse = tensor_to_node_sequence_mapping[node_input]
-            source_tensor = nodes_to_fuse[0].inputs[0]
-            origin_name = source_tensor.split(TENSOR_ASSIGMENT_PREFIX)[0]
-
-            updated_tensor_count[origin_name] += 1
-
-            outputs = [_get_updated_name(origin_name, updated_tensor_count)]
+            if len(node.inputs) == 2:
+                source_tensor = node.inputs[0]
+                outputs = node.outputs
+            else:
+                source_tensor = nodes_to_fuse[0].inputs[0]
+                origin_name = source_tensor.split(TENSOR_ASSIGMENT_PREFIX)[0]
+                updated_tensor_count[origin_name] += 1
+                outputs = [_get_updated_name(origin_name, updated_tensor_count)]
```

Comment:
This is between `mask.fill_(0.2)` with `mask[:, 2] = 0.0`.

```python
class Net(torch.nn.Module):
    def forward(self, x):
        mask = torch.empty(3, 3)
        mask[:, 2] = 0.0
        return mask
```

full raw_graph
```
graph(%self : __torch__.Net,
      %x : Tensor):
  %2 : bool = prim::Constant[value=0]()
  %3 : int = prim::Constant[value=1]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:8:8
  %4 : int = prim::Constant[value=0]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:8:8
  %5 : NoneType = prim::Constant()
  %6 : int = prim::Constant[value=3]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:7:27
  %7 : float = prim::Constant[value=0.]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:8:21
  %8 : int = prim::Constant[value=2]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:8:16
  %9 : int[] = prim::ListConstruct(%6, %6)
  %mask.1 : Tensor = aten::empty(%9, %5, %5, %5, %5, %5) # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:7:15
  %11 : Tensor = aten::slice(%mask.1, %4, %5, %5, %3) # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:8:8
  %12 : Tensor = aten::select(%11, %3, %8) # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:8:8
  %13 : int = prim::dtype(%12)
  %14 : Device = prim::device(%12)
  %15 : Tensor = aten::tensor(%7, %13, %14, %2)
  %16 : Tensor = aten::copy_(%12, %15, %2) # /Users/ryosukefukatani/work/ml-ane-transformers/mask6.py:8:8
  return (%mask.1)
```

```python
class Net(torch.nn.Module):
    def forward(self, x):
        mask = torch.empty(3, 3)
        mask.fill_(0.2)
        return mask
```

full raw_graph
```
graph(%self : __torch__.Net,
      %x : Float(1, 10, 1, 10, strides=[100, 10, 10, 1], requires_grad=0, device=cpu)):
  %2 : int = prim::Constant[value=3]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:8:0
  %3 : int = prim::Constant[value=3]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:8:0
  %4 : int[] = prim::ListConstruct(%2, %3)
  %5 : int = prim::Constant[value=6]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:8:0
  %6 : NoneType = prim::Constant()
  %7 : Device = prim::Constant[value="cpu"]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:8:0
  %8 : bool = prim::Constant[value=0]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:8:0
  %9 : NoneType = prim::Constant()
  %mask : Float(3, 3, strides=[3, 1], requires_grad=0, device=cpu) = aten::empty(%4, %5, %6, %7, %8, %9) # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:8:0
  %11 : float = prim::Constant[value=0.20000000000000001]() # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:10:0
  %12 : Float(3, 3, strides=[3, 1], requires_grad=0, device=cpu) = aten::fill_(%mask, %11) # /Users/ryosukefukatani/work/ml-ane-transformers/mask.py:10:0
  return (%12)
```

In line `  %12 : Tensor = aten::select(%11, %3, %8)` and `  %12 : Float(3, 3, strides=[3, 1], requires_grad=0, device=cpu) = aten::fill_(%mask, %11)`.

1. Getting inputs differently depending on whether the first input is a tensor or not. (In `fill_` case first tensor is tensor,)
2. In `fill_` case, result of `fill_` is used in graph.output. In `mask[:, 2] = 0.0` case `%mask.1` is returned.